src/createHistoryEnhancer.js

```diff
@@ -0,0 +1,27 @@
+import { applyMiddleware } from 'redux';
+
+import createHistoryMiddleware from './createHistoryMiddleware';
+import createTransitionHookMiddleware from './createTransitionHookMiddleware';
+import ensureLocationMiddleware from './ensureLocationMiddleware';
+
+export default function createHistoryEnhancer(protocol, ...middlewares) {
+  return function historyEnhancer(createStore) {
+    return (...args) => {
+      const transitionHookMiddleware = createTransitionHookMiddleware();
+
+      const middlewareEnhancer = applyMiddleware(
```

Comment:
These middlewares should just be the "enhancer" middlewares. Currently the only one is `queryMiddleware` (or `createQueryMiddleware({ stringify, parse })`).

This replaces things like `useQueries`, and instead makes them just be middlewares.